Libs/DICOM/Widgets/ctkDICOMPatientItemWidget.cpp

```diff
@@ -357,20 +366,249 @@ void ctkDICOMPatientItemWidgetPrivate::createStudies()
-  studiesListWidgetLayout->addItem(verticalSpacer);
+  studiesListWidgetLayout->addItem(this->StudiesListVerticalSpacer);
 
   this->IsGUIUpdating = false;
 }
 
+//----------------------------------------------------------------------------
+void ctkDICOMPatientItemWidgetPrivate::updateEnabledServersUIFromDB()
+{
+  if (!this->DicomDatabase)
+  {
+    logger.error("updateEnabledServersUIFromDB, no DICOM Database has been set. \n");
+    return;
+  }
+
+  if (this->PatientItem.isEmpty())
+  {
+    logger.error("updateEnabledServersUIFromDB, PatientItem has not been set. \n");
+    return;
+  }
+
+  if (!this->Scheduler)
+  {
+    logger.error("updateEnabledServersUIFromDB, no scheduler has been set. \n");
+    return;
+  }
+
+  // All active servers (either query/retrieve or storage is toggled).
+  // We assume that the security by default is unknown (patially checked in the UI).
+  // before running any query/retrieve/storage operation the UI needs to ask to the user permissions
+  QStringList allActiveConnectionNames = this->Scheduler->getConnectionNamesForActiveServers();
+  if (allActiveConnectionNames.count() == 0)
+  {
+    this->PatientServersCheckableComboBox->clear();
+    return;
+  }
+
+  // connectionNamesFromDB contains connection names from where the patient has been fetched.
+  // We assume that such server is secure by default
+  QStringList allowList = this->DicomDatabase->enabledConnectionsForPatient(this->PatientItem);
+  QStringList denyList = this->DicomDatabase->disabledConnectionsForPatient(this->PatientItem);
+  QAbstractItemModel* model = this->PatientServersCheckableComboBox->checkableModel();
+  int wasBlocking = this->PatientServersCheckableComboBox->blockSignals(true);
+
+  this->PatientServersCheckableComboBox->clear();
+  this->PatientServersCheckableComboBox->addItems(allActiveConnectionNames);
+  this->EnabledServers.clear();
+  for (int filterIndex = 0; filterIndex < this->PatientServersCheckableComboBox->count(); ++filterIndex)
+  {
+    QString connectionName = this->PatientServersCheckableComboBox->itemText(filterIndex);
+    QModelIndex modelIndex = model->index(filterIndex, 0);
+
+    Qt::CheckState checkState = Qt::CheckState::PartiallyChecked;
+    if (allowList.contains(connectionName))
+    {
+      checkState = Qt::CheckState::Checked;
+    }
+    else if (denyList.contains(connectionName))
+    {
+      checkState = Qt::CheckState::Unchecked;
+    }
+
+    this->PatientServersCheckableComboBox->setCheckState(modelIndex, checkState);
+
+    if (checkState != Qt::CheckState::Unchecked)
+    {
+      this->EnabledServers.append(connectionName);
+    }
+  }
+  this->PatientServersCheckableComboBox->blockSignals(wasBlocking);
+
+  this->saveEnabledServersStringListToChildren(this->EnabledServers);
+}
+
+//----------------------------------------------------------------------------
+bool ctkDICOMPatientItemWidgetPrivate::areAllActiveServersSecure()
```

Comment:
It would be better not to introduce "Secure". We are just talking about allowed and denied servers, allowed does it mean a server is secure; denied does not mean that the server is not secure.
```suggestion
bool ctkDICOMPatientItemWidgetPrivate::areAllActiveServersAllowed()
```